include/cell.tcc

```diff
@@ -1236,6 +1236,27 @@ void mpm::Cell<Tdim>::compute_nodal_momentum(const Eigen::VectorXd& shapefn,
     nodes_[i]->update_momentum(true, phase, shapefn(i) * pmass * pvelocity);
 }
 
+//! Compute Bbar matrix
+template <unsigned Tdim>
+std::vector<Eigen::MatrixXd> mpm::Cell<Tdim>::compute_bbar(
+    const std::vector<Eigen::MatrixXd>& bmatrix, unsigned phase) {
+
+  // Compute BBar
+  std::vector<Eigen::MatrixXd> bbar_matrix;
+  for (unsigned i = 0; i < this->nnodes(); ++i) {
+    Eigen::Matrix<double, 3 * (Tdim - 1), Tdim> bbari;
+    for (unsigned j = 0; j < Tdim; ++j) {
+      for (unsigned k = 0; k < Tdim; ++k)
+        bbari(j, k) =
+            bmatrix.at(i)(j, k) +
+            1. / 3. * (bmatrix_centroid_.at(i)(k, k) - bmatrix.at(i)(k, k));
```

Comment:
Define `OneThird` as a `const double` outside the loop to be `1.0/3.0`

Reply:
Understood